### PR TITLE
Run ogb

### PR DESCRIPTION
**Pull request by ekosman**

### CODE REVIEW

File: src/utils/graph_prune_utils.py [hunk 1]
```diff
@@ -149,11 +149,13 @@ def tg_sample_prune_edges_by_minhash_lsh(tg_sample, minhash, lsh_nodes, lsh_edge
 
 def tg_dataset_prune_edges_by_minhash_lsh(tg_dataset, minhash, lsh_nodes, lsh_edges):
     ratios = list()
+    bar = tqdm(total=len(tg_dataset))
     for i, tg_sample in enumerate(tg_dataset):
```
Comment (joeloren) [inline src/utils/graph_prune_utils.py]: why not simply use `for tg_sample in tqdm(tg_dataset):` . Also, from what I see you're not using `i` anyway.

File: src/utils/graph_prune_utils.py [hunk 2]
```diff
@@ -167,8 +169,10 @@ def tg_sample_prune_random(tg_sample, p, random):
 
 
 def tg_dataset_prune_random(tg_dataset, p, random):
+    bar = tqdm(total=len(tg_dataset))
     for i, tg_sample in enumerate(tg_dataset):
```
Comment (joeloren) [inline src/utils/graph_prune_utils.py]: likewise: `for tg_sample in tqdm(tg_dataset)` (without the `bar.update(1)` line) is simpler.

File: tst/ogb/exp_utils.py [hunk 3]
```diff
@@ -23,11 +26,22 @@ def train(model, device, loader, optimizer, cls_criterion, tb_writer=None):
...
+                if dataset.name == 'ogbg-ppa':
+                    loss = cls_criterion(pred.to(torch.float32),
+                                         batch.y.view(-1,))
+                elif dataset.name in ['ogbg-molhiv', 'ogbg-molpcba']:
+                    # ignore nan targets (unlabeled) when computing training loss.
+                    is_labeled = batch.y == batch.y
+                    # loss = cls_criterion(pred.to(torch.float32)[is_labeled], batch.y.view(-1, ).to(torch.float32)[is_labeled])
+                    # loss = cls_criterion(pred.to(torch.float32)[is_labeled], batch.y.view(-1, )[is_labeled])
+                    # is_labeled_pred = is_labeled.repeat(1, pred.shape[1])
+                    # loss = cls_criterion(pred.to(torch.float32)[is_labeled], batch.y.to(torch.float32)[is_labeled])
```
Comment (joeloren) [inline tst/ogb/exp_utils.py]: Can you get rid of commented-out code?

File: tst/ogb/exp_utils.py [hunk 4]
```diff
@@ -38,60 +52,76 @@ def train(model, device, loader, optimizer, cls_criterion, tb_writer=None):
...
-    elif dataset_name in ['ogbg-code', 'ogbg-ppa']:
+    elif dataset_name in ['ogbg-code2', 'ogbg-ppa']:
         loss = torch.nn.CrossEntropyLoss()
     else:
         raise ValueError("No loss function specified for the given database!")
     return loss
-
-
-def get_tensorboard_logger(log_dir):
-    tb_writer = SummaryWriter(log_dir=log_dir)
```
Comment (joeloren) [inline tst/ogb/exp_utils.py:97]: PEP8 syntax issue. Please run "reformat code" in PyCharm.

File: tst/ogb/main_pyg_with_pruning.py [hunk 5]
```diff
@@ -52,14 +51,12 @@ def get_prune_args(pruning_method: str, num_minhash_funcs: int, random_pruning_p
...
-def main():
-    start_time = time()
-    # Training settings
+def get_args():
     parser = argparse.ArgumentParser(description='GNN baselines on ogbgmol* data with Pytorch Geometrics')
     parser.add_argument('--device', type=int, default=0,
                         help='which gpu to use if any (default: 0)')
     parser.add_argument('--gnn', type=str, default='gcn',
-                        help='GNN gcn, or gcn-virtual (default: gcn)')
+                        help='GNN gcn, or gcn-virtual (default: gcn)', choices=['gcn', ])
```
Comment (joeloren) [inline tst/ogb/main_pyg_with_pruning.py:59]: might as well get rid of this argument if there's only one choice, no?

File: tst/ogb/main_pyg_with_pruning.py [hunk 6]
```diff
@@ -125,41 +130,139 @@ def main():
...
+        if pruning_params is None:
+            dim_nodes = original_dataset[0].x.shape[1] if len(original_dataset[0].x.shape) == 2 else 0
+            lsh_num_funcs = args.num_minhash_funcs
+            sparsity = 2
+            std_of_threshold = 1
+            dim_edges = original_dataset[0].edge_attr.shape[1] if len(original_dataset[0].edge_attr.shape) == 2 else 0
+            minhash = MinHashRep(lsh_num_funcs, random, prime=2147483647)
+            pruning_params = {
+                "minhash": minhash,
+                'nodes': {'din': dim_nodes,
```
Comment (joeloren) [inline tst/ogb/main_pyg_with_pruning.py:183]: `din` -- is this on purpose or a typo?

File: tst/ogb/model_and_data_utils.py [hunk 7]
```diff
@@ -70,9 +72,24 @@ def create_model(dataset: PygGraphPropPredDataset, emb_dim: int, dropout_ratio:
     elif dataset.name == "ogbg-ppa":
         node_encoder = torch.nn.Embedding(1, emb_dim)
         edge_encoder_ctor = partial(torch.nn.Linear, 7)
-        model = GCN(num_classes=dataset.num_tasks, num_layer=num_layers,
-                    emb_dim=emb_dim, drop_ratio=dropout_ratio,
-                    node_encoder=node_encoder, edge_encoder_ctor=edge_encoder_ctor).to(device)
+        # model = GCN(num_classes=dataset.num_classes, num_layer=num_layers,
+        #             emb_dim=emb_dim, drop_ratio=dropout_ratio,
+        #             node_encoder=node_encoder, edge_encoder_ctor=edge_encoder_ctor).to(device)
```
Comment (joeloren) [inline tst/ogb/model_and_data_utils.py]: remove commented-out code.

**Review by joeloren — CHANGES_REQUESTED** [review]

File: tst/ogb/main_pyg_with_pruning.py (hunk 5 above)
Comment (ekosman) [reply tst/ogb/main_pyg_with_pruning.py:59]: We'll add more choices in the future (soon)

File: tst/ogb/main_pyg_with_pruning.py (hunk 6 above)
Comment (ekosman) [reply tst/ogb/main_pyg_with_pruning.py:183]: In purpose. It matches the signature of the LSH class.

**Pull request merged**